enumerables.rb

```diff
@@ -0,0 +1,114 @@
+module Enumerable
+  def my_each()
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    selected_array = []
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      selected_array << self[i] if yield(self[i])
+      i += 1
+    end
+    selected_array
+  end
+
+  def my_all?
+    i = 0
+    while i < size
+      return false if block_given? && !yield(self[i])
+      return false if self[i] == false || self[i].nil?
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    while i < size
+      return true if block_given? && yield(self[i])
+      return true if !self[i] == false && !self[i].nil?
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?
+    i = 0
+    while i < size
+      if block_given?
+        return false if yield(self[i])
+      elsif !self[i] == false && !self[i].nil?
+        return false
+      end
+
+      i += 1
+    end
+    true
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
array = Array.new(100) { rand(0...9) }
array.my_none?(String) == true #true
```
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
words = %w[dog door rod blade]
words.my_none?(5) == words.none?(5) #true
```
- [ ] Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
 words.my_none?(/z/) == words.none?(/z/) #true
 words.my_none?(/d/) == words.none?(/d/) #true
```